storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -219,9 +219,20 @@ public void setBlobMeta(String key, SettableBlobMeta meta, Subject who) throws A
     @Override
     public void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
         validateKey(key);
-        checkForBlobOrDownload(key);
-        SettableBlobMeta meta = getStoredBlobMeta(key);
-        _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+
+        boolean checkPermission = true;
+        try {
+            checkForBlobOrDownload(key);
```

Comment:
Sorry I was mistaken... Even if there's no key on ZK, KeyNotFoundException is not thrown.
KeyNotFoundException is thrown only when there's race condition between ZK node update and removing node from ZK.